From a750aec5c10c6cfd11969bb9d5e557521a43924e Mon Sep 17 00:00:00 2001
From: Lucas Stach <l.stach@pengutronix.de>
Date: Thu, 29 Oct 2015 12:52:46 +0100
Subject: [PATCH 142/162] staging: etnaviv: remove locking from
 etnaviv_gpu_debugfs

It is not necessary as we are only showing register contents here without
relying on GPU state staying constant or modifying any of that state.

The race between one CPU accessing the debugfs file and another one
taking down the GPU is taken care of by the component architecture that
will bring down the DRM master device and the associated debugfs entries
before taking down the GPU.

Signed-off-by: Lucas Stach <l.stach@pengutronix.de>
Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_gpu.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_gpu.c b/drivers/staging/etnaviv/etnaviv_gpu.c
index 43c4117..0160acd 100644
--- a/drivers/staging/etnaviv/etnaviv_gpu.c
+++ b/drivers/staging/etnaviv/etnaviv_gpu.c
@@ -617,10 +617,6 @@ int etnaviv_gpu_debugfs(struct etnaviv_gpu *gpu, struct seq_file *m)
 	if (ret < 0)
 		return ret;
 
-	ret = mutex_lock_interruptible(&gpu->drm->struct_mutex);
-	if (ret < 0)
-		goto err_rpm;
-
 	dma_lo = gpu_read(gpu, VIVS_FE_DMA_LOW);
 	dma_hi = gpu_read(gpu, VIVS_FE_DMA_HIGH);
 	axi = gpu_read(gpu, VIVS_HI_AXI_STATUS);
@@ -721,9 +717,6 @@ int etnaviv_gpu_debugfs(struct etnaviv_gpu *gpu, struct seq_file *m)
 
 	ret = 0;
 
-	mutex_unlock(&gpu->drm->struct_mutex);
-
-err_rpm:
 	pm_runtime_mark_last_busy(gpu->dev);
 	pm_runtime_put_autosuspend(gpu->dev);
 
-- 
2.6.2

